anchore_engine/services/policy_engine/engine/vulns/stores.py

```diff
@@ -108,6 +109,38 @@ def fetch(self, session):
         else:
             return None, None
 
+    def is_modified(self, session, since: datetime.datetime):
+        """
+        Looks up a report for the image and returns True if the report is newer than the timestamp.
+        If the report was generated before the since timestamp, the logic checks for report status and returns True
+        if the report needs a refresh, False otherwise
+
+        """
+        db_record = (
+            session.query(DbImageVulnerabilities)
+            .filter_by(account_id=self.image.user_id, image_digest=self.image.digest)
+            .one_or_none()
```

Comment:
Is there only one DbImageVulnerabilities record for a given digest? Or should this be iterating through a list?  This question might not make sense if I'm totally out of my depth for this feature.

It seems based on the naming like there'd be more records to check. Perhaps more sorting or filtering is required? Perhaps not...

Reply:
Good question, there is only one report per digest per account. That is enforced in the persistence model with a compound primary key consisting of account_id and image_digest